internal/k8s/container.go

```diff
@@ -3,34 +3,35 @@ package k8s
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"github.com/docker/distribution/reference"
 	"github.com/pkg/errors"
 	"github.com/windmilleng/tilt/internal/logger"
 	"k8s.io/api/core/v1"
 )
 
-func WaitForContainerReady(ctx context.Context, client Client, pod *v1.Pod, ref reference.Named) error {
-	ready, err := waitForContainerReadyHelper(pod, ref)
+func WaitForContainerReady(ctx context.Context, client Client, pod *v1.Pod, ref reference.Named) (ContainerID, error) {
+	cID, err := waitForContainerReadyHelper(pod, ref)
 	if err != nil {
-		return err
-	} else if ready {
-		return nil
+		return "", err
+	} else if cID != "" {
+		return cID, nil
 	}
 
 	watch, err := client.WatchPod(ctx, pod)
 	if err != nil {
-		return errors.Wrap(err, "WaitForContainerReady")
+		return "", errors.Wrap(err, "WaitForContainerReady")
 	}
 	defer watch.Stop()
 
 	for true {
```

Comment:
seems like this func renders [`pollForNewClient`](https://github.com/windmilleng/tilt/blob/1b2f93f2f64d24ca19524c725b30089a7a7b5823/internal/engine/sidecar_synclet_manager.go#L69) obsolete, since we only need a single call to this func and it'll do all the waiting and smart decision making for us?

Reply:
it probably doesn't hurt? there might be some other race conditions around opening connections, etc